Subscribers/CronJob.php

```diff
@@ -88,19 +105,60 @@ public function updateProducts(\Shopware_Components_Cron_CronJob $job)
         return true;
     }
 
-    /**
-     * @return ConnectExport
-     */
-    private function getConnectExport()
+    public function exportDynamicStreams(\Shopware_Components_Cron_CronJob $job)
     {
-        return new ConnectExport(
-            $this->getHelper(),
-            $this->getSDK(),
-            Shopware()->Models(),
-            new ProductsAttributesValidator(),
-            $this->getConfigComponent(),
-            new ErrorHandler()
-        );
+        /** @var ProductStreamService $streamService */
+        $streamService = $this->Application()->Container()->get('swagconnect.product_stream_service');
+        $streams = $streamService->getAllExportedStreams(ProductStreamService::DYNAMIC_STREAM);
+
+        /** @var ProductStream $stream */
+        foreach ($streams as $stream) {
+            $streamId = $stream->getId();
+            $productSearchResult = $streamService->getProductFromConditionStream($stream);
+            $orderNumbers = array_keys($productSearchResult->getProducts());
+
+            //no products found
+            if (!$orderNumbers) {
+                //removes all products from this stream
+                $streamService->markProductsToBeRemovedFromStream($streamId);
+            } else {
+                $articleIds = $this->getHelper()->getArticleIdsByNumber($orderNumbers);
+
+                $streamService->markProductsToBeRemovedFromStream($streamId);
+                $streamService->createStreamRelation($streamId, $articleIds);
+
+            }
+            $streamsAssignments = $streamService->prepareStreamsAssignments($streamId, false);
+
+            //article ids must be taken from streamsAssignments
+            $sourceIds = $this->getHelper()->getArticleSourceIds($streamsAssignments->getArticleIds());
+
+            try {
+                $errorMessages = $this->connectExport->export($sourceIds, $streamsAssignments);
```

Comment:
we have to generate delete change for articles which are not part of any shopware stream.